cpp/src/sort/sort_impl.cuh

```diff
@@ -142,9 +144,11 @@ std::unique_ptr<column> sorted_order(table_view input,
                    mutable_indices_view.end<size_type>(),
                    comparator);
     }
+    // protection for temporary d_column_order and d_null_precedence
+    stream.synchronize();
```

Comment:
```suggestion
```

Reply:
The `d_null_precedence` temporary goes out of scope and gets destroyed here.
So this sync is required inside this scope.
The comment was meant to convey that information.